packages/next-server/lib/router/utils/sorted-routes.ts

```diff
@@ -53,12 +58,19 @@ class UrlNode {
         )
       }
 
+      if (this.parentSlugNames.indexOf(slugName) !== -1) {
```

Comment:
Should this be in development only 🤔

Reply:
I'm not following - please could you clarify? 

Did you mean - should this error be triggered only in development?

Reply:
> Did you mean - should this error be triggered only in development?

Yep! I'm not entirely sure though!